src/DeFuncto.Core/Extensions/Tasks.cs

```diff
@@ -50,5 +50,23 @@ async Task<T> WrapInSemaphore(Func<Task<T>> f)
                 }
             }
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Task<T> RunAsync<T>(this T self, Func<T, Task> f) => self.Apply(async t =>
```

Comment:
Since you moved these, move the tests in the test project to match the new structure.